### PR TITLE
Exchanges: Set AccountInfo ID when available

### PR DESCRIPTION
**Pull request by khcchiu**
# PR Description

Set account id when available, changes affecting Binance USDTMarginFutures, Coinbase Spot.

Fixes # (issue)
N/A

## Type of change

Please delete options that are not relevant and add an `x` in `[]` as item is complete.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How has this been tested

Tested Binance/Coinbase by cross-checking api response.

Coinbase Pro is a bit confusing as they have both "profile id" and "account id"

Their response for account balances looks like below:
      #     [
      #         {
      #             id: '4aac9c60-cbda-4396-9da4-4aa71e95fba0',
      #             currency: 'BTC',
      #             balance: '0.0000000000000000',
      #             available: '0',
      #             hold: '0.0000000000000000',
      #             profile_id: 'b709263e-f42a-4c7d-949a-a95c83d065da'
      #         },
      #         {
      #             id: 'f75fa69a-1ad1-4a80-bd61-ee7faa6135a3',
      #             currency: 'USDC',
      #             balance: '0.0000000000000000',
      #             available: '0',
      #             hold: '0.0000000000000000',
      #             profile_id: 'b709263e-f42a-4c7d-949a-a95c83d065da'
      #         },
      #     ]
      #

Which I believe the id is what they call the "account id" but is more like an asset-specific id, and profile_id is what GCT call "ID" in subAccount
reference:
https://forums.coinbasecloud.dev/t/how-do-i-get-account-id-for-rest-api/231/3

Binance reference:
https://dev.binance.vision/t/binance-account-id/1418

I also tried to look at CCXT's implementation and api docs but couldn't find more exchanges that include account id in the response for balances (some returns separately but not from the balance endpoint, some like Binance Spot doesn't give you this piece of info at all as commented in the reference)

- [x] go test ./... -race
- [ ] golangci-lint run
- [ ] Test X

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation and regenerated documentation via the documentation tool
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally and on Github Actions/AppVeyor with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: exchanges/binance/binance_wrapper.go
Comment (shazbert) [inline exchanges/binance/binance_wrapper.go:748]:
```json
Binance raw response: [{"accountAlias":"SgfWmYTiAuoC","asset":"DOT","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"BTC","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"SOL","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"BNB","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"ETH","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"ADA","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"USDT","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"XRP","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0},{"accountAlias":"SgfWmYTiAuoC","asset":"BUSD","balance":"0.00000000","crossWalletBalance":"0.00000000","crossUnPnl":"0.00000000","availableBalance":"0.00000000","maxWithdrawAmount":"0.00000000","marginAvailable":true,"updateTime":0}]
```
This works if you only have one sub-account associated with those keys like the response above but will amalgamate multiple sub-accounts into one. Please change `var acc account.SubAccount` above to `var acc []account.SubAccount` then locally you can make a map[string aka AccountAlias][]account.Balance for example and then you can handle the entirety. 👨‍🎤 


Also out of scope; if you can just check the other asset types while you are there and update appropriately that would be awesome, if not I can check later.

File: exchanges/coinbasepro/coinbasepro_wrapper.go
Comment (shazbert) [inline exchanges/coinbasepro/coinbasepro_wrapper.go:333]: This might be the same issue as above, might need to handle different profile ID's to balances which can be independently added to repsonse.Accounts.

**Review by shazbert — CHANGES_REQUESTED** [review]
Thanks for opening this PR!, Unfortunately while testing, I do not have multiple accounts associated with my API keys for those exchanges at the moment. Let me know what you think of the requested changes. 🚀

File: exchanges/binance/binance_wrapper.go
Comment (khcchiu) [reply exchanges/binance/binance_wrapper.go]: Yes make sense, added changes to separate balances into different buckets according to account id and a helper function to collect them back into a slice of accounts.

File: exchanges/coinbasepro/coinbasepro_wrapper.go
Comment (khcchiu) [reply exchanges/coinbasepro/coinbasepro_wrapper.go]: Yes, applied the changes as well, also on Bitmex.

**Review by shazbert — CHANGES_REQUESTED** [review]
Thanks for addressing my concerns; the new function for accounts looks great! Just a pedantic nit. 

Binance looks good: 
```
INFO: [{SgfWmYTiAuoC usdtmarginedfutures [{DOT 0 0} {BTC 0 0} {SOL 0 0} {BNB 0 0} {ETH 0 0} {ADA 0 0} {USDT 0 0} {XRP 0 0} {BUSD 0 0}]}]
```

I don't have keys atm for bitmex or coinbasepro, but I have cross referenced the API docs.

File: exchanges/account/account.go
Comment (khcchiu) [reply exchanges/account/account.go]: Sure, renamed and added checks.

**Review by shazbert — APPROVED** [review]
Thanks for making those changes and opening this PR. 🚀  Will wait on another two confirmations before merge. 

tACK. 
```
{Exchange:Binance Accounts:[{ID:SgfWmYTiAuoC AssetType:usdtmarginedfutures Currencies:[{CurrencyName:DOT TotalValue:0 Hold:0} {CurrencyName:BTC TotalValue:0 Hold:0} {CurrencyName:SOL TotalValue:0 Hold:0} {CurrencyName:BNB TotalValue:0 Hold:0} {CurrencyName:ETH TotalValue:0 Hold:0} {CurrencyName:ADA TotalValue:0 Hold:0} {CurrencyName:USDT TotalValue:0 Hold:0} {CurrencyName:XRP TotalValue:0 Hold:0} {CurrencyName:BUSD TotalValue:0 Hold:0}]} {ID: AssetType:usdtmarginedfutures Currencies:[]}]}
```

File: exchanges/binance/binance_wrapper.go
Comment (gloriousCode) [inline exchanges/binance/binance_wrapper.go:745]: Could you also do this for case USDTMarginedFutures too?

**Review by gloriousCode — CHANGES_REQUESTED** [review]
Hey @khcchiu ! Thanks for opening this PR. Just one thing from me so far

File: exchanges/binance/binance_wrapper.go
Comment (khcchiu) [reply exchanges/binance/binance_wrapper.go:745]: Hi, the case added with USDTMarginatedFutures, is it also available for CoinMarginatedFutures?

but for coin the model is below which does not include account alias:
`// FuturesAccountInformation stores account information for futures account
type FuturesAccountInformation struct {
	Assets []struct {
		Asset                  string  `json:"asset"`
		WalletBalance          float64 `json:"walletBalance,string"`
		UnrealizedProfit       float64 `json:"unrealizedProfit,string"`
		MarginBalance          float64 `json:"marginBalance,string"`
		MaintMargin            float64 `json:"maintMargin,string"`
		InitialMargin          float64 `json:"initialMargin,string"`
		PositionInitialMargin  float64 `json:"positionInitialMargin,string"`
		OpenOrderInitialMargin float64 `json:"openOrderInitialMargin,string"`
		MaxWithdrawAmount      float64 `json:"maxWithdrawAmount,string"`
		CrossWalletBalance     float64 `json:"crossWalletBalance,string"`
		CrossUnPNL             float64 `json:"crossUnPnl,string"`
		AvailableBalance       float64 `json:"availableBalance,string"`
	} `json:"assets"`
	Positions   []FuturesAccountInformationPositions `json:"positions"`
	CanDeposit  bool                                 `json:"canDeposit"`
	CanTrade    bool                                 `json:"canTrade"`
	CanWithdraw bool                                 `json:"canWithdraw"`
	FeeTier     int64                                `json:"feeTier"`
	UpdateTime  time.Time                            `json:"updateTime"`
}`

Comment (gloriousCode) [reply exchanges/binance/binance_wrapper.go:745]: Apologies! I said the wrong asset type.
I meant to do the following for CoinMarginedFutures
```
case asset.CoinMarginedFutures:
		accData, err := b.GetFuturesAccountBalance(ctx)
		if err != nil {
			return info, err
		}
		accountCurrencyDetails := make(map[string][]account.Balance)
		for i := range accData {
			currencyDetails := accountCurrencyDetails[accData[i].AccountAlias]
			accountCurrencyDetails[accData[i].AccountAlias] = append(
				currencyDetails, account.Balance{
					CurrencyName: currency.NewCode(accData[i].Asset),
					TotalValue:   accData[i].Balance,
					Hold:         accData[i].Balance - accData[i].AvailableBalance,
				},
			)
		}

		if info.Accounts, err = account.CollectBalances(accountCurrencyDetails, assetType); err != nil {
			return account.Holdings{}, err
		}
```

**Review by gloriousCode — COMMENTED** [review]
Cannot verify Bitmex at the moment, however it matches API documentation. Binance and Coinbase both work. Let me know how you go with CoinMarginedFutures addition, but otherwise, everything here is good

**Comment by khcchiu** [comment]
> Cannot verify Bitmex at the moment, however it matches API documentation. Binance and Coinbase both work. Let me know how you go with CoinMarginedFutures addition, but otherwise, everything here is good

For current implementation, []assets from /dapi/v1/account is being used for coin margined futures which does not include account alias
https://binance-docs.github.io/apidocs/delivery/en/#account-information-user_data

Though I see there's another endpoint /dapi/v1/balance, which does include account alias
https://binance-docs.github.io/apidocs/delivery/en/#futures-account-balance-user_data

Do you think if /dapi/v1/balance is equivalent to /dapi/v1/account's []assets? However unfortunately my Binance account is registered with a region not allowing derivatives trading so I likely couldn't access the /dapi.

**Review by gloriousCode — APPROVED** [review]
> However unfortunately my Binance account is registered with a region not allowing derivatives trading so I likely couldn't access the /dapi

Then I'm definitely not going to request that you implement it 😄 tACK! Thanks for your contribution!

**Review by thrasher- — APPROVED** [review]
utACK for Bitmex/CoinbasePro, code looks good! Thanks @khcchiu

**Pull request merged**
